File: atcoder.jp/Python (CPython 3.11.4)/abc318/abc318_c.py
N, D, P = map(int, input().split())
F = list(map(int, input().split()))
F.sort(reverse=True)

acc = [0] * (N + 1)
for i in range(N):
    acc[i + 1] = acc[i] + F[i]

ans = float("inf")
i = 0
while True:
    x = i * P
    n = i * D
    tmp = acc[N] - acc[min(n, N)] + x
    ans = min(ans, tmp)
    i += 1
    if n >= N:
        break

print(ans)

File: atcoder.jp/Rust (rustc 1.70.0)/abc318/abc318_c.rs
use proconio::{input, fastout};

#[inline]
fn ceil(a: usize, b: usize) -> usize {
    (a + b - 1) / b
}

#[fastout]
fn main() {
    input! {
        n: usize,
        d: usize,
        p: usize,
        mut f: [usize; n],
    }

    f.sort();
    f.reverse();
    let mut acc = vec![0; n + 1];
    for i in 0..n {
        acc[i + 1] = acc[i] + f[i];
    }

    let mut ans = usize::MAX;
    for i in 0..=ceil(n, d) {
        let l = i * d;
        ans = ans.min((acc[n] - acc[l.min(n)]) + p * i);
    }

    println!("{}", ans);
}
